pennylane/ops/op_math/controlled.py

```diff
@@ -556,11 +559,29 @@ def _decompose_no_control_values(op: "operation.Operator") -> List["operation.Op
     no decomposition.
     """
     if len(op.control_wires) == 1 and hasattr(op.base, "_controlled"):
-        return [op.base._controlled(op.control_wires[0])]
-    if isinstance(op.base, qml.PauliX):
-        if len(op.control_wires) == 2:
-            return [qml.Toffoli(op.active_wires)]
+        result = op.base._controlled(op.control_wires[0])
+        # disallow decomposing to itself
+        # pylint: disable=unidiomatic-typecheck
+        if type(result) != type(op):
+            return [result]
+        qml.QueuingManager.remove(result)
+    if (
+        isinstance(op.base, qml.PauliX)
+        or len(op.base.wires) == 1
+        and getattr(op.base, "has_matrix", False)
+        and qmlmath.get_interface(*op.data) == "numpy"
+        and qmlmath.allclose(op.base.matrix(), qml.PauliX.compute_matrix())
```

Comment:
```suggestion
```
This might have some strange consequences.  We could consider making this a task in itself, but I'd need to think about whether or not this is desired behavior.